store/store_download.go

```diff
@@ -48,12 +48,20 @@ import (
 	"github.com/snapcore/snapd/snapdtool"
 )
 
-var downloadRetryStrategy = retry.LimitCount(7, retry.LimitTime(90*time.Second,
-	retry.Exponential{
-		Initial: 500 * time.Millisecond,
-		Factor:  2.5,
-	},
-))
+// Retry strategies use internal state, in particular LimitTime uses time.Now()
+// as a starting point, therefore this needs to be a factory function and not
+// a plain variable.
+var downloadRetryStrategy = func() retry.Strategy {
+	// XXX: LimitTime doesn't make much sense with downloads, since
+	// download time counts towards time limit, making it give up on retries
+	// too early with slow downloads.
```

Comment:
This is know, added a comment so we don't forget. I can look at it in near future.